redis_session.c

```diff
@@ -140,6 +140,27 @@ static int session_gc_maxlifetime(void) {
     return value;
 }
 
+/* Retrieve redis.session.compression from php.ini */
+static int session_compression_type(void) {
+#ifdef HAVE_REDIS_LZF
+    if(strcmp(INI_STR("redis.session.compression"), "lzf") == 0) {
+        return REDIS_COMPRESSION_LZF;
+    }
+#endif
+#ifdef HAVE_REDIS_ZSTD
+    if(strcmp(INI_STR("redis.session.compression"), "zstd") == 0) {
+        return REDIS_COMPRESSION_ZSTD;
+    }
+#endif
+#ifdef HAVE_REDIS_LZ4
+    if(strcmp(INI_STR("redis.session.compression"), "lz4") == 0) {
+        return REDIS_COMPRESSION_LZ4;
+    }
+#endif
+
```

Comment:
Maybe it's worth issuing an E_NOTICE `php_error_docref` here if the INI setting is not empty and not "none"?